endpoints/openrtb2/video_auction.go

```diff
@@ -79,7 +81,27 @@ func (deps *endpointDeps) VideoAuctionEndpoint(w http.ResponseWriter, r *http.Re
 		CookieFlag:    pbsmetrics.CookieFlagUnknown,
 		RequestStatus: pbsmetrics.RequestStatusOK,
 	}
+
+	debugQuery := r.URL.Query().Get("debug")
+	enableDebug := debugQuery == "true"
+	var debugString string
+	var errorCacheID string
+
 	defer func() {
+		if len(errorCacheID) > 0 && vo.VideoResponse == nil {
+			cache := deps.ex.GetCache()
```

Comment:
Asking the exchange for an instance of the cache doesn't make sense to me. I recommend making the cache available directly to the video auction endpoint handler. We use a simple DI construct in router.go. Then you don't need to add a method to the exchange interface... and you can then also pre-check that it isn't nil.

Reply:
Updated to have the cache directly available to the video endpoint. It is instead passed in from the router. Also added nil check for the cache.

Reply:
Thank you. :)

Reply:
No problem!